### PR TITLE
Minor updates to ch5 based on LRV's review

### PR DESCRIPTION
**Pull request by jni**

### CODE REVIEW

File: markdown/ch5.markdown [hunk 1]
```diff
@@ -430,32 +430,27 @@ print('The CSR and NumPy arrays are equal: ',
...
 one can think of the entire web as a large, sparse, $N \times N$ matrix.
 Each entry $X_{ij}$ indicates whether web page $i$ links to page $j$.
 By normalizing this matrix and solving for its dominant eigenvector,
 one obtains the so-called PageRank—one of the numbers Google uses to
 order your search results. (You can read more about this in the next chapter!)
 
-Now, consider studying the human brain, and represent it as a large $M
-\times M$ graph, where there are $M$ nodes (positions) in which you
+As another example, we can represent the human brain as a large $m \times m$
+graph, where there are $m$ nodes (positions) in which you
```
Comment (stefanv) [inline markdown/ch5.markdown]: the connection between connectivity matrix and graph is never made

File: markdown/ch5.markdown [hunk 2]
```diff
@@ -430,32 +430,27 @@ print('The CSR and NumPy arrays are equal: ',
...
-regions of the brain!
-
-[^submod]: One has to be careful when subdividing the network into
-           more than one group.  The graph under analysis remains the
-           original, *not* the previously split result.  A correct
-           approach is explained in detail in Newman (2006).
+Thresholding this matrix produces a sparse matrix of ones and zeros. 
+The eigenvector corresponding to the second-smallest eigenvalue of this matrix
+partitions the $m$ brain areas into subgroups, which, it turns out,
+correspond to functional regions of the brain [^Newman]!
```
Comment (stefanv) [inline markdown/ch5.markdown]: Hrm, not sure if that is true.  It's most likely not true.

File: markdown/ch5.markdown [hunk 3]
```diff
@@ -577,14 +572,12 @@ were head of the NumPy Agency for Space Affairs and had to rotate
 millions of images streaming in from the newly launched Jupyter
 Orbiter?
 
-Of course, one option would be to rewrite all your code in C++, but
-let's presume you have diligently read Scott Meyers' excellent books
-and are deeply convinced that you will never write bug free C++ under
-pressure.  While options like Cython, Numba, or Julia are available,
-we'd like to show you a quick workaround using SciPy's sparse
-matrices.
+In such cases, you want to squeeze every ounce of performance from your
```
Comment (stefanv) [inline markdown/ch5.markdown]: I don't recall the TR comment, but why is this paragraph better than the previous version?

File: markdown/ch5.markdown [hunk 4]
```diff
@@ -1394,7 +1389,7 @@ tiger = io.imread(url)
 plt.imshow(tiger);
 ```
 
-In order to check our image segmentation, we're going to need a ground truth.
+In order to check our image segmentation, we're going to need ground truth.
```
Comment (stefanv) [inline markdown/ch5.markdown]: a -> the?

http://www.dictionary.com/browse/ground-truth

File: markdown/ch5.markdown [hunk 5]
```diff
@@ -1407,7 +1402,9 @@ But to be clear, we really have no single ground truth!
 from scipy import ndimage as ndi
 from skimage import color
 
-human_seg_url = 'http://www.eecs.berkeley.edu/Research/Projects/CS/vision/bsds/BSDS300/html/images/human/normal/outline/color/1122/108073.jpg'
+human_seg_url = ('http://www.eecs.berkeley.edu/Research/Projects/CS/'
```
Comment (stefanv) [inline markdown/ch5.markdown]: This syntax is valid, but may confuse new Python users.  Worth adding a '+'?

File: markdown/ch5.markdown (hunk 2 above)
Comment (jni) [inline markdown/ch5.markdown]: Look at ch6. Either you do the second-smallest eigenvalue, or you have to make changes to the matrix to make it a "modularity matrix" — which is no longer sparse!

Though come to think of it even for 2nd smallest you need to divide each row by its degree. =\ Should we maybe just scrap this example?

File: markdown/ch5.markdown (hunk 3 above)
Comment (jni) [inline markdown/ch5.markdown]: The comment was that the intro was overly complicated. My own take is that

a) a very tiny minority of our readers would have thought of Cython/Numba/Julia, and a much tinier still minority would have thought of C++. So it felt a bit jarring to me to mention all that.
b) at any rate, SciPy's functions are implemented highly optimised C, so it's unclear why we should be able to speed them up.
c) we never mentioned that the thing we can make fast is repeated application of the same transform. Now we do.

File: markdown/ch5.markdown (hunk 4 above)
Comment (jni) [inline markdown/ch5.markdown]: How about "some"?

File: markdown/ch5.markdown (hunk 5 above)
Comment (jni) [inline markdown/ch5.markdown]: I understand your concern, but it's self-explanatory after spending a second to think about it, and most readers (I hope) would go on to try it in their open IPython console and learn something new! So I'd prefer to leave this as is. I used to type the + all the time until I realised it was redundant!

File: markdown/ch5.markdown (hunk 2 above)
Comment (jni) [inline markdown/ch5.markdown]: I think the example ends up with too many caveats, so I'm definitely leaning towards scrapping it.

Comment (stefanv) [inline markdown/ch5.markdown]: Dividing doesn't remove sparsity, does it?

File: markdown/ch5.markdown (hunk 3 above)
Comment (stefanv) [inline markdown/ch5.markdown]: OK, that's fine with me.

File: markdown/ch5.markdown (hunk 4 above)
Comment (stefanv) [inline markdown/ch5.markdown]: Yup!

File: markdown/ch5.markdown (hunk 5 above)
Comment (stefanv) [inline markdown/ch5.markdown]: :)  Learning on the job, OK!

File: markdown/ch5.markdown (hunk 2 above)
Comment (stefanv) [inline markdown/ch5.markdown]: Oh, and also: I think I should have stated more clearly what I meant.

"correspond to functional regions of the brain" == False

Comment (jni) [inline markdown/ch5.markdown]: No, it doesn't remove sparsity but it removes simplicity. =) it no longer becomes easy enough to state the problem in a paragraph with no caveats. 

How would you phrase the functionality of the resulting modules? Again, if we couch the statements with too many caveats, it becomes pretty useless as an in-passing example.

**Comment by jni** [comment]
@stefanv I've remove the example. Perhaps it can be included in Ch6 in a future edition. ;)

**Comment by stefanv** [comment]
@jni I thought it was a great example; I just wanted a small change like from "it turns out that these correspond to functional brain areas" to "it turns out that these are often related to functional brain areas".

**Comment by jni** [comment]
@stefanv here you go. =)

File: markdown/ch5.markdown (hunk 1 above)
Comment (stefanv) [inline markdown/ch5.markdown]: I still think it would be worth saying something here about edge == correlation strength.

**Comment by jni** [comment]
@stefanv please push to my branch or merge + edit

**Pull request merged**